### PR TITLE
Fix compatibility with non-GNU tar

### PR DESCRIPTION
**Pull request by dcl**
**Fix compatibility with non-GNU tar**

Currently, several `tar` commands in the singularity cli use the `-B` option, a GNU extension. This breaks on host systems with non-GNU tar. Removing this option resolves the issue, without changing the behavior of GNU tar -- all instances of this option are used when reading from stdin, which will [cause GNU tar to automatically enable this option](https://www.gnu.org/software/tar/manual/tar.html#SEC73).

**This fixes or addresses the following GitHub issues:**

- None

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [x] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [x] I have tested this PR locally with a `make test`
- [x] This PR is NOT against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

**Review by GodloveD — APPROVED** [review]
I ran this through a bunch of tests and it seems to work great.  I approve if somebody else does too.

**Review by ArangoGutierrez — APPROVED** [review]

**Comment by ArangoGutierrez** [comment]
Looks good to me

**Comment by GodloveD** [comment]
@dcl do you have time to fix the conflicts in the `CONTRIBUTORS.md` file?  If so, I think we can merge right away!

**Comment by dcl** [comment]
@GodloveD done.

**Comment by GodloveD** [comment]
Thank you @dcl !

**Pull request merged**
